virttest/qemu_devices/qdevices.py

```diff
@@ -2243,6 +2243,22 @@ def is_direct_plug(self, device):
 
         return True
 
+    def _get_port_addr(self, root_port):
+        """
+        Get root port address for function port
+
+        :param root_port: pcie-root-port QDevice object
+        :return string: pcie-root-port address or None if slot is full
+        """
+        slot = root_port.get_param('addr').split('.')[0]
+        full_addrs = set(['%s.%s' % (slot, hex(_)) for _ in range(1, 8)])
+        used_addrs = set(self.__root_ports.keys())
+        try:
+            return sorted(list(full_addrs - used_addrs))[0]
+        except IndexError:
+            logging.debug('slot %s is full' % slot)
```

Comment:
Got this flooded the debug.log, let's consider to remove it.
```
2020-02-11 14:47:32,200 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,201 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,212 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,212 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,212 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,213 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,213 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,213 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,213 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,213 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,228 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,228 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,228 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,229 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,229 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,229 qdevices         L2259 DEBUG| slot 0x1 is full
2020-02-11 14:47:32,229 qdevices         L2259 DEBUG| slot 0x1 is full
...
```
Another thing is that is not that effective if it is called so many times. Since we still have many other tasks to do, let's currently leave the refactoring work to the future, thanks.